docs/products/postgresql/concepts/upgrade-failover.rst

```diff
@@ -57,3 +57,9 @@ During maintenance updates, cloud migrations, or plan changes, the below procedu
 
 .. Note::
     The old primary server is kept alive for a short period of time (minimum 60 seconds) with a TCP forwarding setup pointing to the new primary server allowing clients to connect before learning the new IP address.
+
+Recreation of replication slots
+-------------------------------
+
+In case of failover or controlled switchover of an Aiven for PostgreSQL service, the replication slots from the old primary server are recreated in the new primary server by default.
```

Comment:
Can we get a bit more info on what's happening here? I believe this deviates from the standard PostgreSQL feature so we need to tell people a bit more what we're doing under the hood

Reply:
Replication slots are recovered after failover on the new primary automatically. This is a new behaviour after https://github.com/aiven/aiven-core/pull/26663 
I would avoid word `default` as there is no other non-default behaviour now.

How it works (not sure if everything should be put to the public doc):
1. If we have only one node cluster, the new node before replacing the old one will get info about replication slots on the original service and will re-create them and will then fail over.
2. For multi-node setups we synchronize the replication slots from the primary to standbys periodically.
    * every 30 (we might change it, so probably should not put a specific number here) seconds we ensure that for newly created slots the dependencies are installed in the corresponding databases
    * every several seconds we synchronize the positions (confirmed_flush_lsn) of the slots on the primary and standbys
    * if failover happens to the standby, it will have replication slots with almost up-to-date positions from the primary (position of the replication slots on the new primary might be around 5 seconds behind the positions of old primary)

Therefore in case of uncontrolled failover slots created up to 30 seconds before failover might be lost.
Due to the specifics of implementation, the position of the recovered replication slot might be several seconds older then on the primary, so in case of connecting to the slot after the failover without specifying position, the client might receive old entities which has already received before.